src/oidc/index.js

```diff
@@ -0,0 +1,64 @@
+import OIDCStrategy from './strategy';
+
+const DEFAULT_URL_PATTERN = /auth\/oidc\/callback/;
+
+class OIDCAuth {
+  constructor(options = {}) {
+    this.options = options;
+    this.strategy = new OIDCStrategy(options);
+    this.callbackURLPattern = options.callbackURLPattern || DEFAULT_URL_PATTERN;
+  }
+
+  init() {
+    // URL may be updated before now and the strategy.init promise resolution...
+    const currentUrl = window.location.href;
+
+    return this.strategy.init()
+      .then(() => {
```

Comment:
We're using Babel, so can we use async await here?

Reply:
This is a global decision we did not make for now